libs/isograph-react/src/loadable-hooks/useConnectionSpecPagination.ts

```diff
@@ -20,10 +20,27 @@ import { useIsographEnvironment } from '../react/IsographEnvironmentProvider';
 import { useSubscribeToMultiple } from '../react/useReadAndSubscribe';
 import { maybeUnwrapNetworkRequest } from '../react/useResult';
 
-type FirstOrAfter = 'first' | 'after';
-type OmitFirstAfter<TArgs> = keyof Omit<TArgs, FirstOrAfter> extends never
-  ? void | Record<string, never>
-  : Omit<TArgs, FirstOrAfter>;
+type First = 'first';
+type After = 'after';
+
+type MakeAfterOptional<TArgs> = {
+  [K in Extract<keyof TArgs, After>]?: TArgs[K] | undefined;
+} & {
+  [K in Exclude<keyof TArgs, After>]: TArgs[K];
+};
+
+type Prettify<T> = {
+  readonly [K in keyof T]: T[K];
+} & {};
+
+// if none of the keys are required initialArgs will be optional
+type MakeOptionalArgs<TArgs> = {} extends TArgs
+  ? [initialArgs?: TArgs]
```

Comment:
`void | TArgs` requires passing `undefined`, `[initialArgs?: TArgs]` allows for no extra argument

Reply:
neat trick!

Reply:
we can use arguments[1] to not allocate array, and then cast the hook to type the variable 🤔